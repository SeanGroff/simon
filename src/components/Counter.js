// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  power: boolean,
  counter: ?number,
};

const StyledCounter = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border: 4px solid #222222;
  background-color: #32050c;
  font-size: 32px;
  border-radius: 10px;
  width: 50px;
  color: ${({ power }: Props) => (power ? '#DC0D29' : '#430710')};
`;

export default function Counter({ counter, power }: Props) {
  return <StyledCounter power={power}>{counter || '--'}</StyledCounter>;
}
