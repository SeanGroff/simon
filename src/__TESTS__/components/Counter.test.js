import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../../components/Counter';

it('renders a counter display', () => {
  const component = shallow(<Counter />);
  expect(component).toMatchSnapshot();
});
