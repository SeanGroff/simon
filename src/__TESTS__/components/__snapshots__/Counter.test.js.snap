// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a counter display 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    count={0}
>
    --
</styled.div>,
  "nodes": Array [
    <styled.div
      count={0}
>
      --
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Counter
        count={0}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "count": 0,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          count={0}
>
          --
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          count={0}
>
          --
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Counter
    count={0}
/>,
}
`;
